src/plugin.scss

```diff
@@ -25,3 +25,20 @@ Styles for ad integrations.
 .vjs-ad-playing .vjs-audio-button {
     display: none;
 }
+// only animate when showing because it can be processor heavy
+.vjs-ad-playing.vjs-ad-loading .vjs-loading-spinner:before,
```

Comment:
I believe you want `vjs-ad-loading` not `vjs-ad-playing`.

Besides that this just needs a comment explaining where the code is copied from, and what we want to do for a long-term fix.